### PR TITLE
Helm Chart for Hashicorp Vault.

### PR DESCRIPTION
**Pull request by keyolk**
Helm Chart for Hashicorp Vault.
Supports Consul and InMem backend.
Support TLS.

### CODE REVIEW

File: vault/templates/vault.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+{{- if .Values.Vault.Tls }}
+        volumeMounts:
+        - name: config
+          mountPath: /etc/vault/cfg
+        - name: tls
+          mountPath: /etc/vault/tls
+{{- end }}
+        env:
+        - name: "VAULT_DEBUG"
+          value: "true"
```
Comment (coffeepac) [inline vault/templates/vault.yaml]: do we always want this set to true?

File: vault/values.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Vault:
+  Image: "quay.io/keyolk/vault"
```
Comment (coffeepac) [inline vault/values.yaml]: can you push this image to quay.io/samsung_cnct/vault?  then its under group control (if you lack the ability to do this, ping me and I'll check what's up).

and is there a repo somewhere I can look at for how this image gets built?  if not, there should be a samsung-cnct repo (samsung-cnct/docker-vault or something) that has the dockerfile, a make file and a README for how to build this image.

Comment (keyolk) [inline vault/values.yaml]: OK, I'll update it.

File: vault/templates/vault.yaml (hunk 1 above)
Comment (keyolk) [inline vault/templates/vault.yaml]: I think it is better to enable this, to check what happen in this component.
Unless hard to debug.

Comment (coffeepac) [inline vault/templates/vault.yaml]: how much information will vault emit on debug?  I'm thinking about this from a disk-pressure point of view, ie:  how long will this component run with logging on debug before the disk fills up and the pod restarts.

Comment (keyolk) [inline vault/templates/vault.yaml]: not to much.
for example ...

```
2016/12/27 23:17:38 create third party resource "auth-code.oidc.coreos.com"
2016/12/27 23:17:38 create third party resource "auth-request.oidc.coreos.com"
2016/12/27 23:17:38 create third party resource "o-auth2-client.oidc.coreos.com"
2016/12/27 23:17:38 create third party resource "signing-key.oidc.coreos.com"
2016/12/27 23:17:38 create third party resource "refresh-token.oidc.coreos.com"
2016/12/27 23:17:38 create third party resource "password.oidc.coreos.com"
2016/12/27 23:17:38 keys expired, rotating
2016/12/27 23:17:38 failed to rotate keys: not found
2016/12/27 23:17:38 listening (https) on 0.0.0.0:443
2016/12/27 23:17:40 http: TLS handshake error from 10.128.228.0:18973: EOF
2016/12/27 23:18:08 keys expired, rotating
2016/12/27 23:18:08 keys rotated, next rotation: 2016-12-28 05:18:08.852269069 +0000 UTC
2016/12/28 00:28:43 http: TLS handshake error from 10.128.228.0:21043: EOF
2016/12/28 00:29:27 http2: server: error reading preface from client 10.128.88.0:33183: timeout waiting for client preface
2016/12/28 05:18:13 keys expired, rotating
2016/12/28 05:18:13 keys rotated, next rotation: 2016-12-28 11:18:13.152754315 +0000 UTC
2016/12/28 11:18:17 keys expired, rotating
2016/12/28 11:18:17 keys rotated, next rotation: 2016-12-28 17:18:17.550604317 +0000 UTC
2016/12/28 17:18:22 keys expired, rotating
2016/12/28 17:18:22 keys rotated, next rotation: 2016-12-28 23:18:22.609321583 +0000 UT
```

above is my logs from k2
for a day.

Comment (coffeepac) [inline vault/templates/vault.yaml]: okay.  DEBUG is fine then.  thanks!

File: vault/values.yaml (hunk 2 above)
Comment (keyolk) [inline vault/values.yaml]: the image pushed to https://quay.io/repository/samsung_cnct/consul
Dockerfile pushed to https://github.com/samsung_cnct/consul

Comment (coffeepac) [inline vault/values.yaml]: Thanks!

**Comment by coffeepac** [comment]
LGTM.

**Pull request merged**